components/relevancy/src/lib.rs

```diff
@@ -159,8 +161,76 @@ impl RelevancyStore {
     /// its likelihood of a negative outcome.
     #[handle_error(Error)]
     pub fn bandit_update(&self, bandit: String, arm: String, selected: bool) -> ApiResult<()> {
+        let mut cache = self.cache.lock();
+
+        cache
+            .clear(&bandit, &arm)
+            .expect("failed to clear cached distribution");
+
         self.db
             .read_write(|dao| dao.update_bandit_arm_data(&bandit, &arm, selected))?;
+
+        Ok(())
+    }
+}
+
+pub struct BanditCache {
```

Comment:
Let's do this instead if `impl Default` ourselves:

`#[derive(Default)]
pub struct BanditCache`